aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,769 @@
+
+        Returns
+        -------
+        numpy.dtype
+            The data format used to store the data in the Leica lif file and the read numpy.ndarray.
+        """
+        return self.dask_data.dtype
+
+    @property
+    def metadata(self) -> Element:
+        """
+        Load and return the metadata from the LIF file
+
+        Returns
+        -------
+        The lxml Element Tree of the metadata
+        """
+        # We can't serialize lxml element trees so don't save the tree to the object state
+        meta_xml, header = utilities.get_xml(self._file)
+        return meta_xml
+
+    def _size_of_dimension(self, dim: str) -> int:
+        if dim in self.dims:
+            return self.dask_data.shape[self.dims.index(dim)]
+
+        return 1
+
+    def size_s(self) -> int:
+        return self._size_of_dimension(Dimensions.Scene)
+
+    def size_t(self) -> int:
+        return self._size_of_dimension(Dimensions.Time)
+
+    def size_c(self) -> int:
+        return self._size_of_dimension(Dimensions.Channel)
+
+    def size_z(self) -> int:
+        return self._size_of_dimension(Dimensions.SpatialZ)
+
+    def size_y(self) -> int:
+        return self._size_of_dimension(Dimensions.SpatialY)
+
+    def size_x(self) -> int:
+        return self._size_of_dimension(Dimensions.SpatialX)
+
+    def get_channel_names(self, scene: int = 0) -> List[str]:
+        """
+        Get the channel names for the scene
+
+        Parameters
+        ----------
+        scene: int
+            The index of the scene from which to retrieve the channel names
+
+        Returns
+        -------
+        List[str]
+            A list of descriptive names of the channels of the form "Gray--TL-BF--EMP_BF" and "Green--FLUO--GFP"
+        """
+        # for the remove to work the
+        img_sets = self.metadata.findall(".//Image")
+
+        img = img_sets[scene]
+        scene_channel_list = []
+        chs = img.findall(".//ChannelDescription")
+        chs_details = img.findall(".//WideFieldChannelInfo")
+        for ch in chs:
+            ch_detail = next(x for x in chs_details if x.attrib["LUT"] == ch.attrib["LUTName"])
+            scene_channel_list.append((f"{ch_detail.attrib['LUT']}--{ch_detail.attrib['ContrastingMethodName']}"
+                                       f"--{ch_detail.attrib['FluoCubeName']}"))
+        return scene_channel_list
+
+    # TODO refactor this utility function into a metadata wrapper class
+    def _getmetadataxmltext(self, findpath, default=None):
+        ref = self.metadata.find(findpath)
+        if ref is None:
+            return default
+        return ref.text
+
+    def get_physical_pixel_size(self, scene: int = 0) -> Tuple[float]:
+        """
+        Get the (X, Y, Z) pixel size. If the value isn't set it returns the 1.0.
+
+        Parameters
+        ----------
+        scene: int
+            The scene to retrieve the dimensions from
+
+        Returns
+        -------
+        (X, Y, Z) in µm.
```

Comment:
It is always micrometers? Or is that dependent on the scope configuration?

Reply:
Can't know without more files. Currently, they're all in um.

Reply:
I can have it return m rather than um if that's our standard unit.

Reply:
I am not really suggesting a change to this without re-assessing all the other readers :(

Reply:
I looked at the other readers and realized that I had implemented that code so I had chosen to return um not meters. I made it consistent with the others so now it returns m.